app/controllers/lead_providers/your_schools_controller.rb

```diff
@@ -12,18 +12,21 @@ def index
                            @cohorts.find_by(start_year: Time.zone.today.year)
                          end
 
-      @schools = School.eligible.partnered_with_lead_provider(@lead_provider.id, @selected_cohort.start_year)
-        .includes(:early_career_teachers)
-        .order(:name)
+      @partnerships = Partnership
+        .includes(school: :early_career_teachers)
+        .order("schools.name")
+        .where(
+          cohort: @selected_cohort,
+          lead_provider: @lead_provider,
+        )
```

Comment:
I might have misunderstood, but I think we're missing an edge-case here. 

1. LP reports partnership with school
2. School declares that partnership is incorrect
3. LP reports partnership with school again

We end up with two partnerships in this instance, one challenged and one not. Would the school appear twice in this list?

Reply:
Ah, excellent question! I'd imagine we should just update the existing partnership instead of creating a new record?

Reply:
No, we need to retain an easy record of the challenge

Reply:
Otherwise answering a question like "how often are challenged partnerships reinstated" becomes quite tricky. And that's something we'll be interested in

Reply:
Not sure what the best way to do this would be. If you used [active_record_distinct_on](https://github.com/alecdotninja/active_record_distinct_on) it could become `Partnership.distinct_on(:school_id).order(:school_id, challenged_at: :desc)`

Reply:
Doesn't seem very robust though, `.count` wouldn't work

Reply:
Ok, I *think* it would be great for complexity here to decouple the current state of the partnership from the history of that partnership. If the partnership is reinstated after the challenge, I'd argue this is still the same challenge (not to say that a unique index on [:school_id, :lead_provider_id] would be quite a nice constraint to have). Once the school is reconfirmed, than the old, challenged partnership has no domain meaning. If we were to look at this with Event Sourcing, all the Partnership::Reported and Partnership::Challenged events would live in a single stream.

There are quite a few options here - one being having duplicated partnerships. But this makes things really complicated in terms of getting the list of partnerships. Honestly, it feels a bit strange that we are displaying the challenged partnership in the "your schools" list - it is not really their school when it is challenged.

Another option - papertrail the partnerships. We could add extra scopes to find the challenge versions so that the stats are visible.

And option 3 - database Log record. (Polymorphing) object belonging to any record in the system representing major changes in the history of that object. It is a bit like a reversed event sourcing, where we are still in monolithic state of "now" as a source of truth, but with some history points.

I'd vote for option 3!

Reply:
I don't see how 3 differs from papertrail? Is that not a description of the Versions table?

Reply:
OR we propose e.g. having a Your Schools that is _actually_ your schools, and a separate view for all partnership challenges? That is not incompatible with any of the other options here, but seems to make the most sense as an interface, right?